### PR TITLE
bug fixing on issue/8/piecewise linear quantiles

### PR DESCRIPTION
**Pull request by johannct**
The pad quantiles were not coded correctly, as far as I can tell, and there was a bug in the utils package : a flattening call missing for a returned array

### CODE REVIEW

File: qp/utils.py [hunk 1]
```diff
@@ -362,9 +362,10 @@ def interpolate_multi_x_multi_y(x, xvals, yvals, **kwargs):
     xy_vals = np.hstack([xvals, yvals])
     nx = xvals.shape[-1]
     def single_row(xy_vals_):
+        print(xy_vals_[0:nx].shape,xy_vals_[nx:].shape)
         return interp1d(xy_vals_[0:nx], xy_vals_[nx:], **kwargs)(x)
     vv = np.vectorize(single_row, signature="(%i)->(%i)" % (xvals.shape[-1]+yvals.shape[-1], x.size))
-    return vv(xy_vals)
+    return  vv(xy_vals).flatten()
```
Comment (johannct) [inline qp/utils.py:367]: @eacharles I had to add the flattening here to pass the unit test. And the flattening for the factored case is present in the hist version of this code, previously used in quant_pdf. Still, can you cross-check me here, and if correct check the docstrings of the rest of the code, as I think these are copy-paste all the way down

File: qp/quant_pdf.py [hunk 2]
```diff
@@ -53,18 +53,17 @@ def pad_quantiles(quants, locs):
         pad_hi = False
     if n_out == n_vals:
         return quants, locs
-    ratio = (locs[:, -1] - locs[:, 0]) / (quants[-1] - quants[0])
     quants_out = np.zeros((n_out), quants.dtype)
     locs_out = np.zeros((locs.shape[0], n_out), quants.dtype)
     quants_out[offset_lo:n_vals+offset_lo] = quants
     locs_out[:,offset_lo:n_vals+offset_lo] = locs
```
Comment (johannct) [inline qp/quant_pdf.py:59]: @aimalz : this below is what I understand the extrapolation on the cdf function should look like, in order to pad it to 0 on the left and 1 on the right. I did not put the epsilon everywhere though, maybe we should?

File: qp/utils.py (hunk 1 above)
Comment (eacharles) [reply qp/utils.py:367]: I should probably sit down at some point and figure out exactly what is going on with the array shaping and flattening and write it all down.  I agree that it is confusing.   But this looks good for now.

**Review by eacharles — APPROVED** [review]
This is fine by me, maybe you want to remove the unused commented printout and debuggin lines are some point, but feel free to leave them in as long as you think they are useful.

**Pull request merged**